crates/wasi/src/lib.rs

```diff
@@ -1,272 +1,23 @@
-//! # Wasmtime's WASI Implementation
```

Comment:
Mind keeping some of the crate comment here? My rough goal has been to make the [landing page](https://docs.rs/wasmtime-wasi/latest/wasmtime_wasi/) relatively informative. No need to duplicate p1/p2 docs, but having pointers/links to those modules and an outline of the high-level organization here I think would be good.